src/comm/MAVLinkProtocol.cc

```diff
@@ -198,6 +210,33 @@ void MAVLinkProtocol::receiveBytes(LinkInterface* link, QByteArray b)
                 if (!(mavlinkStatus->flags & MAVLINK_STATUS_FLAG_IN_MAVLINK1) && (mavlinkStatus->flags & MAVLINK_STATUS_FLAG_OUT_MAVLINK1)) {
                     qDebug() << "Switching outbound to mavlink 2.0 due to incoming mavlink 2.0 packet:" << mavlinkStatus << mavlinkChannel << mavlinkStatus->flags;
                     mavlinkStatus->flags &= ~MAVLINK_STATUS_FLAG_OUT_MAVLINK1;
+
+                    // Setup mavlink 2 signing
+                    if (_appSettings->mavlink2Signing()->rawValue().toBool()) {
+                        mavlink_setup_signing_t setupSigning;
+                        QString key = _appSettings->mavlink2SigningKey()->rawValue().toString();
+
+                        setupSigning.target_system = message.sysid;
+                        setupSigning.target_component = message.compid;
+                        setupSigning.initial_timestamp = QDateTime(QDate(2015, 1, 1)).msecsTo(QDateTime::currentDateTimeUtc()) * 100;
```

Comment:
By inspecting this with the debugger I get a huge value for the starting time stamp: 8032606717800
I personally don't know if it is correct.